src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,104 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int LENGTH = 10;
+
+    private T [] defaultArray;
+    private int size;
+
+    public ArrayList() {
+        this.defaultArray = (T[]) new Object[LENGTH];
+        this.size = 0;
+    }
+
+    private boolean isFull() {
+        if (size >= defaultArray.length) {
+            T [] largeArray = (T []) new Object [(int)(defaultArray.length * 1.5)];
+            System.arraycopy(defaultArray, 0, largeArray, 0, size);
+            defaultArray = largeArray;
+        }
+        return false;
+    }
 
     @Override
     public void add(T value) {
-
+        isFull();
+        defaultArray[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index > size) {
+            throw new ArrayIndexOutOfBoundsException("Can't add element with this index");
+        }
+        T[] newArray = (T[]) new Object[size + 1];
```

Comment:
let's not create the new array
you can use defaultArray as both source and destination in your System.arraycopy

Reply:
Fixed